source/features/default-branch-button.tsx

```diff
@@ -39,6 +39,11 @@ async function add(branchSelector: HTMLElement): Promise<void> {
 		</a>
 	);
 
+	// Don't show the button if the file is not exists on the default branch
+	if (await is404Page(url.href)) {
```

Comment:
On second though, I don't want this button to be delayed on every page load, so we should always add it regardless of 404 status. We can drop this condition

Reply:
I'll check is404 only on RepoFile404 page.